--- xulrunner-1.9.1.4.orig/js/src/nanojit/Assembler.cpp
+++ xulrunner-1.9.1.4/js/src/nanojit/Assembler.cpp
@@ -50,8 +50,20 @@
 #endif
 
 #ifdef AVMPLUS_SPARC
+#ifdef __linux__
+void sync_instruction_memory(caddr_t v, u_int len)
+{
+	caddr_t end = v + len;
+	caddr_t p = v;
+	while (p < end) {
+		asm volatile("flush %0" : : "r" (p));
+		p += 32;
+	}
+}
+#else
 extern  "C"	void sync_instruction_memory(caddr_t v, u_int len);
 #endif
+#endif
 
 namespace nanojit
 {