--- libavformat/os_support.c	2010-01-15 23:29:54.000000000 +0100
+++ libavformat/os_support.c	2010-01-15 23:30:11.000000000 +0100
@@ -30,7 +30,7 @@
 #include <sys/time.h>
 #include "os_support.h"
 
-#if CONFIG_NETWORK
+//#if CONFIG_NETWORK
 #if !HAVE_POLL_H
 #if HAVE_WINSOCK2_H
 #include <winsock2.h>
@@ -220,7 +220,7 @@ int ff_socket_nonblock(int socket, int e
       return fcntl(socket, F_SETFL, fcntl(socket, F_GETFL) & ~O_NONBLOCK);
 #endif
 }
-#endif /* CONFIG_NETWORK */
+//#endif /* CONFIG_NETWORK */
 
 #if CONFIG_FFSERVER
 #if !HAVE_POLL_H
