--- ./scripts/src/osd/sdl_cfg.lua	2016-08-31 17:21:29.975886157 +0200
+++ ./scripts/src/osd/sdl_cfg.lua	2016-08-31 17:22:08.741865452 +0200
@@ -117,14 +117,16 @@
 	configuration { }
 
 elseif _OPTIONS["targetos"]=="linux" then
-	if _OPTIONS["QT_HOME"]~=nil then
-		buildoptions {
-			"-I" .. backtick(_OPTIONS["QT_HOME"] .. "/bin/qmake -query QT_INSTALL_HEADERS"),
-		}
-	else
-		buildoptions {
-			backtick(pkgconfigcmd() .. " --cflags Qt5Widgets"),
-		}
+	if _OPTIONS["USE_QTDEBUG"]=="1" then
+		if _OPTIONS["QT_HOME"]~=nil then
+			buildoptions {
+				"-I" .. backtick(_OPTIONS["QT_HOME"] .. "/bin/qmake -query QT_INSTALL_HEADERS"),
+			}
+		else
+			buildoptions {
+				backtick(pkgconfigcmd() .. " --cflags Qt5Widgets"),
+			}
+		end
 	end
 elseif _OPTIONS["targetos"]=="macosx" then
 	defines {
