--- Jamroot	2009-08-28 13:52:51.000000000 +0200
+++ Jamroot	2009-08-28 13:53:01.000000000 +0200
@@ -352,7 +352,7 @@
         if $(layout) = versioned
         {
             result = [ common.format-name
-                <base> <toolset> <threading> <runtime> -$(BOOST_VERSION_TAG)
+                <base> <threading> <runtime> -$(BOOST_VERSION_TAG)
                 -$(BUILD_ID)
                 : $(name) : $(type) : $(property-set) ] ;
         }        
