--- boost/spirit/home/lex/lexer/token_def.hpp
+++ boost/spirit/home/lex/lexer/token_def.hpp
@@ -109,7 +109,12 @@
         template <typename Context>
         info what(Context& /*context*/) const
         {
-            return info("token_def");
+            if (0 == def_.which()) {
+                return info("token_def", get<string_type>(def_));
+            }
+            else {
+                return info("token_def", get<char_type>(def_));
+            }
         }
 
         ///////////////////////////////////////////////////////////////////////
