--- boost/spirit/home/qi/stream/stream.hpp	2013-01-05 20:45:14.659666875 +0100
+++ boost/spirit/home/qi/stream/stream.hpp	2013-01-05 20:45:27.528273334 +0100
@@ -66,7 +66,7 @@
             qi::skip_over(first, last, skipper);
 
             instream in(first, last);           // copies 'first'
-            in >> attr_;                        // use existing operator>>()
+            in >> attr;                         // use existing operator>>()
 
             // advance the iterator if everything is ok
             if (in) {
