--- Jamroot.old	2009-04-08 23:31:15.000000000 +0200
+++ Jamroot	2008-11-03 16:50:31.000000000 +0100
@@ -323,7 +323,7 @@
         if $(layout) = versioned
         {
             local result = [ common.format-name
-                <base> <toolset> <threading> <runtime> -$(BOOST_VERSION_TAG)
+                <base> <threading> <runtime> -$(BOOST_VERSION_TAG)
                 -$(BUILD_ID)
                 : $(name) : $(type) : $(property-set) ] ;
             
