--- src/mumble/Log.cpp	2009-06-26 15:42:57.000000000 +0200
+++ src/mumble/Log.cpp	2009-06-26 15:42:25.000000000 +0200
@@ -32,6 +32,9 @@
 #include "TextToSpeech.h"
 #include "MainWindow.h"
 #include "Global.h"
+#ifdef USE_DBUS
+#include "DBus.h"
+#endif
 
 #ifdef Q_OS_MAC
 extern bool qt_mac_execute_apple_script(const QString &script, AEDesc *ret);
@@ -283,6 +286,7 @@
 		g.mw->qteLog->ensureCursorVisible();
 	}
 
+#ifdef USE_DBUS
 	// Message notification with balloon tooltips
 	if ((flags & Settings::LogBalloon) && !(g.mw->isActiveWindow() && g.mw->qdwLog->isVisible()))  {
 		QString qsIcon;
@@ -362,6 +366,7 @@
 			qt_mac_execute_apple_script(qsScript, NULL);
 #endif
 	}
+#endif
 
 	// Message notification with static sounds
 
