--- branches/gcc-4_6-branch/gcc/cp/decl2.c	2011/06/23 16:53:03	175342
+++ branches/gcc-4_6-branch/gcc/cp/decl2.c	2011/06/23 16:56:27	175343
@@ -4285,6 +4285,9 @@
   if (TREE_CODE (decl) == FUNCTION_DECL
       && DECL_NONSTATIC_MEMBER_FUNCTION_P (decl)
       && DECL_DEFAULTED_FN (decl)
+      /* A function defaulted outside the class is synthesized either by
+	 cp_finish_decl or instantiate_decl.  */
+      && !DECL_DEFAULTED_OUTSIDE_CLASS_P (decl)
       && ! DECL_INITIAL (decl))
     {
       /* Remember the current location for a function we will end up
