--- mk/cmake/Modules/FindCURL.cmake	2011-04-29 13:03:15.000000000 +0200
+++ mk/cmake/Modules/FindCURL.cmake	2011-04-29 13:04:09.000000000 +0200
@@ -30,19 +30,6 @@
     FIND_PROGRAM( CMAKE_CURL_CONFIG curl-config)
 
     IF(CMAKE_CURL_CONFIG)
-      # run the curl-config program to get --static-libs
-      EXEC_PROGRAM(sh
-	ARGS "${CMAKE_CURL_CONFIG} --static-libs"
-        OUTPUT_VARIABLE CURL_STATIC_LIBS
-        RETURN_VALUE RET)
-
-      MESSAGE(STATUS "CURL RET = ${RET}")
-
-      IF(${RET} EQUAL 0)
-        MESSAGE(STATUS "USING CURL STATIC LIBS: ${CURL_STATIC_LIBS}")
-      	SET(CURL_LIBRARIES "-Bstatic ${CURL_STATIC_LIBS}")
-      ELSE()
-
         EXEC_PROGRAM(sh
 	  ARGS "${CMAKE_CURL_CONFIG} --libs"
           OUTPUT_VARIABLE CURL_STATIC_LIBS
@@ -51,7 +38,6 @@
         MESSAGE(STATUS "#2 CURL RET = ${RET} libs: ${CURL_STATIC_LIBS}")
         SET(CURL_LIBRARIES "${CURL_STATIC_LIBS}")
 
-      ENDIF()
     ENDIF()
 ENDIF()
 ELSE(CURL_FOUND)
