--- xpdf-3.0.2-poppler/xpdf.cc	2009-10-19 17:36:28.000000000 +0200
+++ xpdf-3.0.2-poppler/xpdf.cc	2009-10-19 17:36:59.000000000 +0200
@@ -27,7 +27,6 @@ static char enableT1libStr[16] = "";
 static char enableFreeTypeStr[16] = "";
 static char antialiasStr[16] = "";
 static char vectorAntialiasStr[16] = "";
-static char autohintingStr[16] = "no";
 static char psFileArg[256];
 static char paperSize[15] = "";
 static int paperWidth = 0;
@@ -80,8 +79,6 @@ static ArgDesc argDesc[] = {
    "enable font anti-aliasing: yes, no"},
   {"-aaVector",   argString,      vectorAntialiasStr, sizeof(vectorAntialiasStr),
    "enable vector anti-aliasing: yes, no"},
-  {"-ah",         argString,      autohintingStr, sizeof(autohintingStr),
-   "enable font auto-hinting: yes, no (default no)"},
   {"-ps",         argString,      psFileArg,      sizeof(psFileArg),
    "default PostScript file name or command"},
   {"-paper",      argString,      paperSize,      sizeof(paperSize),
@@ -212,16 +209,6 @@ int main(int argc, char *argv[]) {
       fprintf(stderr, "Bad '-aaVector' value on command line\n");
     }
   }
-  if (autohintingStr[0]) {
-    GBool ah;
-    if (!globalParamsGUI->parseYesNo2(autohintingStr, &ah)) {
-      fprintf(stderr, "Bad '-ah' value on command line\n");
-    }
-    if (ah == gTrue)
-      globalParams->setForceNoFTAutoHinting("no");
-    else
-      globalParams->setForceNoFTAutoHinting("yes");
-  }
   if (printCommands) {
     globalParamsGUI->setPrintCommands(printCommands);
   }
