--- acinclude.m4.old	2012-01-26 19:10:17.732969900 +0100
+++ acinclude.m4	2012-01-26 19:10:33.482477277 +0100
@@ -1411,7 +1411,7 @@
 if test x$ax_python_bin != x; then
    AC_CHECK_LIB($ax_python_bin, main, ax_python_lib=$ax_python_bin, ax_python_lib=no)
    AC_CHECK_HEADER([$ax_python_bin/Python.h],
-   [[ax_python_header=`locate $ax_python_bin/Python.h | sed -e s,/Python.h,,`]],
+   [[ax_python_header=/usr/include/$ax_python_bin]],
    ax_python_header=no)
    if test x$ax_python_lib != xno; then
      if test x$ax_python_header != xno; then
