--- MultiMC5-0.6.0/application/CMakeLists.txt	2018-01-12 20:25:29.015824162 +0100
+++ MultiMC5-0.6.0/application/CMakeLists.txt	2018-01-12 20:25:45.278832225 +0100
@@ -404,7 +404,7 @@
 
 	MESSAGE(STATUS "Compiling for linux system with ${MultiMC_SHARE_DEST_DIR} and MULTIMC_LINUX_DATADIR")
 	set_target_properties(MultiMC PROPERTIES OUTPUT_NAME ${MultiMC_APP_BINARY_NAME})
-	target_compile_definitions(MultiMC PRIVATE "-DMULTIMC_JARS_LOCATION=/${MultiMC_SHARE_DEST_DIR}/jars" "-DMULTIMC_LINUX_DATADIR"
+	target_compile_definitions(MultiMC PRIVATE "-DMULTIMC_JARS_LOCATION=${CMAKE_INSTALL_PREFIX}/${MultiMC_SHARE_DEST_DIR}/jars" "-DMULTIMC_LINUX_DATADIR"
 	)
 
 	# install as bundle with no dependencies included
