--- configure.ac	2014-04-26 12:43:35.640817606 +0200
+++ configure.ac	2014-04-26 12:38:51.961857363 +0200
@@ -430,8 +430,8 @@
 #  error not gcc 4
 #endif
 
-#if __GNUC_MINOR__ != 4 && __GNUC_MINOR__ != 5 && __GNUC_MINOR__ != 6 && __GNUC_MINOR__ != 7 && __GNUC_MINOR__ != 8
-#  error not gcc 4.4 to 4.8
+#if __GNUC_MINOR__ != 4 && __GNUC_MINOR__ != 5 && __GNUC_MINOR__ != 6 && __GNUC_MINOR__ != 7 && __GNUC_MINOR__ != 8 && __GNUC_MINOR__ != 9
+#  error not gcc 4.4 to 4.9
 #endif
 ])],
 	[AC_MSG_RESULT([yes])
