--- media/ffmpeg/ffmpeg_common.h.old	2011-02-19 12:47:31.000000000 +0100
+++ media/ffmpeg/ffmpeg_common.h	2011-02-19 12:46:59.000000000 +0100
@@ -17,7 +17,7 @@
 // TODO(scherkus): fix and upstream the compiler warnings.
 MSVC_PUSH_DISABLE_WARNING(4244);
 #include <libavcodec/avcodec.h>
-#include <libavcore/samplefmt.h>
+#include <libavutil/samplefmt.h>
 #include <libavformat/avformat.h>
 #include <libavformat/avio.h>
 #include <libavutil/avutil.h>
