diff -up mozilla-1.9.1/widget/src/gtk2/nsNativeKeyBindings.cpp.gtk mozilla-1.9.1/widget/src/gtk2/nsNativeKeyBindings.cpp
--- mozilla-1.9.1/widget/src/gtk2/nsNativeKeyBindings.cpp.gtk	2009-07-15 14:23:24.000000000 +0200
+++ mozilla-1.9.1/widget/src/gtk2/nsNativeKeyBindings.cpp	2009-07-15 14:23:29.000000000 +0200
@@ -42,7 +42,7 @@
 #include "nsGtkKeyUtils.h"
 #include "nsGUIEvent.h"
 
-#include <gtk/gtkentry.h>
+#include <gtk/gtk.h>
 #include <gtk/gtktextview.h>
 #include <gtk/gtkbindings.h>
 #include <gtk/gtkmain.h>
