--- src/common/mm_multi_file_io.cpp.old	2011-02-02 11:42:14.000000000 +0100
+++ src/common/mm_multi_file_io.cpp	2011-02-02 11:43:03.000000000 +0100
@@ -15,6 +15,7 @@
 
 #include <algorithm>
 #include <boost/regex.hpp>
+#include <boost/version.hpp>
 #include <vector>
 
 #include "common/mm_multi_file_io.h"
@@ -159,7 +160,11 @@ mm_multi_file_io_c::display_other_file_i
 
   foreach(const mm_multi_file_io_c::file_t &file, m_files)
     if (file.m_file_name != m_files.front().m_file_name)
+#if BOOST_VERSION >= 104600
+      file_names.push_back(file.m_file_name.leaf().string());
+#else
       file_names.push_back(file.m_file_name.leaf());
+#endif
 
   if (!file_names.empty())
     mxinfo(boost::format(Y("'%1%': Processing the following files as well: %2%\n")) % m_display_file_name % join(", ", file_names));
