--- src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h	2012-03-23 15:28:44.778842586 +0100
+++ src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSGlobalObject.h	2012-03-23 15:34:50.538586443 +0100
@@ -57,7 +57,6 @@
     
     class JSGlobalObject : public JSVariableObject {
     protected:
-        using JSVariableObject::JSVariableObjectData;
 
         struct JSGlobalObjectData : public JSVariableObjectData {
             // We use an explicit destructor function pointer instead of a
--- src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h	2012-03-23 15:32:52.776207855 +0100
+++ src/3rdparty/javascriptcore/JavaScriptCore/runtime/JSStaticScopeObject.h	2012-03-23 15:33:01.906927075 +0100
@@ -32,7 +32,6 @@
     
     class JSStaticScopeObject : public JSVariableObject {
     protected:
-        using JSVariableObject::JSVariableObjectData;
         struct JSStaticScopeObjectData : public JSVariableObjectData {
             JSStaticScopeObjectData()
                 : JSVariableObjectData(&symbolTable, &registerStore + 1)
