--- libslirc2/trunk/src/auxiliary_string_md5.cpp	(revision 33)
+++ libslirc2/trunk/src/auxiliary_string_md5.cpp	(working copy)
@@ -23,6 +23,7 @@
 #include "auxiliary_string.hpp"
 
 #include <cstring>
+#include <stdio.h>
 
 namespace slIRC {
 namespace string {
--- libslirc2/trunk/CMakeLists.txt	(revision 33)
+++ libslirc2/trunk/CMakeLists.txt	(working copy)
@@ -1,7 +1,22 @@
+FIND_PACKAGE(
+	Boost 1.35.0 COMPONENTS
+	regex 
+	thread
+)
+
+IF(NOT Boost_FOUND)
+	MESSAGE(ERROR "Boost not found")
+	return()
+ENDIF()
+
+INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
+
+LINK_DIRECTORIES(${Boost_LIBRARY_DIRS})
+
 FILE (GLOB files src/*.cpp)
 ADD_LIBRARY (slirc2 SHARED ${files})
 FILE (GLOB slircheaders src/*.h*)
-TARGET_LINK_LIBRARIES (slirc2 boost_regex boost_thread)
+TARGET_LINK_LIBRARIES (slirc2 ${Boost_LIBRARIES})
 INSTALL (FILES ${slircheaders} DESTINATION
   ${CMAKE_INSTALL_PREFIX}/include/slirc)
-INSTALL (TARGETS slirc2 DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
\ No newline at end of file
+INSTALL (TARGETS slirc2 DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
