--- plugins/LadspaEffect/calf/src/modules_limit.cpp.old	2016-11-02 19:35:17.592311880 +0100
+++ plugins/LadspaEffect/calf/src/modules_limit.cpp	2016-11-02 19:36:01.576370393 +0100
@@ -540,7 +540,7 @@
             } // process single strip with filter
 
             // write multiband coefficient to buffer
-            buffer[pos] = std::min(*params[param_limit] / std::max(fabs(sum_left), fabs(sum_right)), 1.0);
+            buffer[pos] = std::min(*params[param_limit] / std::max(fabs(sum_left), fabs(sum_right)), 1.0f);
 
             for (int i = 0; i < strips; i++) {
                 // process gain reduction
