diff --git a/src/XExtInt.c b/src/XExtInt.c
index a5e70b5..cd6cc45 100644
--- a/src/XExtInt.c
+++ b/src/XExtInt.c
@@ -268,7 +268,7 @@ static XExtensionVersion versions[] = { {XI_Absent, 0, 0},
  XI_Add_DevicePresenceNotify_Minor},
 {XI_Present, XI_Add_DeviceProperties_Major,
  XI_Add_DeviceProperties_Minor},
-{XI_Present, XI_2_Major, XI_2_Minor}
+{XI_Present, XI_2_Major, 0}
 };
 
 /***********************************************************************
