diff -Nurp CGAL-3.4-old/CMakeLists.txt CGAL-3.4/CMakeLists.txt
--- CGAL-3.4-old/CMakeLists.txt	2009-09-21 17:22:24.317923913 +0200
+++ CGAL-3.4/CMakeLists.txt	2009-09-21 17:22:50.201563400 +0200
@@ -118,6 +118,8 @@ if( MSVC )
     
 endif()
 
+ADD_DEFINITIONS("-DCORE_NO_AUTOMATIC_NAMESPACE")
+
 if( "${CMAKE_CXX_COMPILER_ID}" MATCHES SunPro )
   message( STATUS "Using SunPro compiler, using STLPort 4." )
   
