--- Graphics/UI/SDL/Events.hsc.old	2014-05-04 10:17:49.590418122 +0200
+++ Graphics/UI/SDL/Events.hsc	2014-05-04 10:24:13.799733621 +0200
@@ -53,7 +53,7 @@
 import qualified Prelude (Enum(..))
 
 import Foreign.StablePtr (newStablePtr,castStablePtrToPtr,castPtrToStablePtr,deRefStablePtr)
-import Data.Typeable (Typeable(typeOf),TypeRep)
+import Data.Typeable (typeOf,Typeable,TypeRep)
 
 import Graphics.UI.SDL.Keysym (SDLKey, Modifier, Keysym)
 import Graphics.UI.SDL.Utilities (Enum(..), intToBool, toBitmask, fromBitmask, fromCInt)
